### DISCUSSION TITLE
Need help in Prisma Logging

### DISCUSSION THREAD

**Discussion by rujhanArora** [prisma/prisma#12061] [Q&A] [answered]
Hi Prisma Team, Need your help with the following-
1. How can I use my custom Logger for logging queries? We have extended the Winston logger and customized it. How can we use that logger to log queries?
2. How can I log only a particular type of query? I want to only log Update/Create queries, but not select queries.
3. How can I add some custom fields in the log statement? For example, I have the userId of the person who performed a query. How to log that userId along with the query?
Looking forward to hearing from you. Thank you!
CC: @janpio

**Comment by ajmnz** [comment, marked as answer]
You can log queries by listening to the `query` event.

```ts
const prisma = new PrismaClient();

prisma.$on("query", (e) => {
  Logger.debug("Query: " + e.query);
  Logger.debug("Duration: " + e.duration + "ms");
});
```

I don't think you have access to the query type, so maybe you could try to filter them with something like `e.query.includes("UPDATE")`

**Reply by rujhanArora** [reply]
Thanks mate! This answers the first 2 issues. Can you also help me with the 3rd one? Looking forward to hearing from you!

**Reply by ajmnz** [reply]
> Can you also help me with the 3rd one?

Unless you have some kind of global state within your application, I think your best bet is to wrap the queries with a function that will log whatever you pass to it.

```ts
const withLog = <T>(cb: () => T, userId: number): T => {
  // This will get logged right before Prisma emits the `query` event
  // you're listening to with `prisma.$on("query", ...)`
  Logger.debug(`Query by user ${userId}`);
  return cb();
};

const user = await withLog(() => prisma.customer.findFirst(), 1234);
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
